File: Count_Number_of_Distinct_Integers_After_Reverse_Operations/Count_Number_of_Distinct_Integers_After_Reverse_Operations.py

nums = [1,13,10,12,31]


def countDistinctIntegers(nums):
    s2 = set()
    s1 = set(nums)
    for i in nums:
        str1 = str(i)
        s1.add(int(str1[::-1]))

    return len(s1)

print(countDistinctIntegers(nums))

File: Count_Number_of_Distinct_Integers_After_Reverse_Operations/Count_Number_of_Distinct_Integers_After_Reverse_Operations.go
package main

import (
	"fmt"
	"strconv"
)

func main() {
	nums := []int{1, 13, 10, 12, 31}
	fmt.Println(countDistinctIntegers(nums))
}

func countDistinctIntegers(nums []int) int {
	res := make(map[int]struct{})
	for _, num := range nums {
		res[num] = struct{}{}
		str1 := strconv.Itoa(num)
		rev, _ := strconv.Atoi(reverse(str1))
		res[int(rev)] = struct{}{}
	}


	return len(res)
}

func reverse(str1 string) (out string) {
	for _, s := range str1 {
		out = string(s) + out
	}

	return out
}
